### PR TITLE
PXP-10542 Client creds rotation test

### PR DESCRIPTION
**Pull request by paulineribeyre**
Jira Ticket: [PXP-10542](https://ctds-planx.atlassian.net/browse/PXP-10542)

### New Features
- Client creds rotation test

[PXP-10542]: https://ctds-planx.atlassian.net/browse/PXP-10542?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: services/apis/fence/fenceTasks.js [hunk 1]
```diff
@@ -48,6 +48,20 @@ module.exports = {
...
+   * Runs a fence command to rotate a client's ID and secret
+   * @param {string} clientName - client name
+   */
+  rotateClientCredentials(clientName, expiresInDays) {
+    let fenceCmd = `fence-create client-rotate --client ${clientName}`;
+    if (expiresInDays) {
+      fenceCmd = `${fenceCmd} --expires-in ${expiresInDays}`;
+    }
+    const res = bash.runCommand(fenceCmd, 'fence', takeLastLine);
+    const arr = res.replace(/[()']/g, '').split(',').map((val) => val.trim());
```
Comment (haraprasadj) [inline services/apis/fence/fenceTasks.js:61]: possible to add an example of res in the docstrings for making it easy to understand the goal of this line for newcomers?

File: suites/apis/oidcClientTest.js [hunk 2]
```diff
@@ -59,3 +59,32 @@ Scenario('OIDC Client Expiration @clientExpiration', async ({ I, fence }) => {
...
+  const creds2 = fence.do.rotateClientCredentials(clientName);
+  expect(creds2.client_id, 'No client ID').not.to.be.empty;
+  expect(creds2.client_secret, 'No client secret').not.to.be.empty;
+  if (process.env.DEBUG === 'true') {
+    console.log(`Client credentials after rotating: (${creds2.client_id}, ${creds2.client_secret})`);
+  }
+
+  // check that both sets of credentials work (we can get an access token using the creds)
+  console.log('Checking credentials obtained before rotating...');
+  await fence.do.getAccessTokenWithClientCredentials(creds1.id, creds1.secret);
```
Comment (haraprasadj) [inline suites/apis/oidcClientTest.js:87]: do we expect successful access token creation with older credentials after the rotation is done? or, should we add expectSuccess=false to this call?

Comment (paulineribeyre) [reply suites/apis/oidcClientTest.js:87]: yes, this is what the feature is about, both should work (rotation without downtime). there are more details in the ticket description and linked design document

**Review by haraprasadj — APPROVED** [review]
+1

**Review by atharvar28 — APPROVED** [review]

**Comment by paulineribeyre** [comment]
ignoring failures of currently broken tests - merging

**Pull request merged**